gradle-with-plugin/build.gradle

```
@@ -0,0 +1,137 @@
+import com.github.dynamicextensionsalfresco.gradle.tasks.InstallBundle
+
+plugins {
+    id 'java'
+    id 'idea'
+
+    // Plugin used to be able to use the 'alfrescoProvided' dependency scope. 
+    // See: https://github.com/xenit-eu/alfresco-gradle-sdk#the-euxenitalfresco-plugin
+    id "eu.xenit.alfresco" version "0.2.0"
+
+    // Plugin used to assemble the project into an Alfresco Module Package
+    // See: https://github.com/xenit-eu/alfresco-gradle-sdk#the-euxenitamp-plugin
+    id "eu.xenit.amp" version "0.2.0"
+
+    // Plugin used to start an Alfresco for testing purposes
+    // See: https://github.com/xenit-eu/alfresco-docker-gradle-plugin
+    id "eu.xenit.docker-alfresco" version "4.0.3"
+
+    // The Dynamic Extensions Gradle plugin
+    // See: https://github.com/xenit-eu/dynamic-extensions-for-alfresco/blob/master/documentation/Gradle_Plugin.md
+    id "eu.xenit.de" version "2.0.1"
+}
+
+
+description = 'Dynamic Extensions Example Web Scripts, Actions and Behaviours.'
+version = '2.0-SNAPSHOT'
+
+ext {
+    dynamicExtensionsVersion = '2.0.1'
+
+    dynamicExtensionsAlfrescoVersion = "61"
+    alfrescoVersion = '6.1.2-ga'
+    shareVersion = "6.1.0"
+}
+
+repositories {
+    mavenCentral()
+    maven {
+        url "https://artifacts.alfresco.com/nexus/content/groups/public/"
+    }
+}
+
+configurations {
+    // Custom configuration added to demonstrate a custom InstallBundle task which can upload OSGi dependencies 
+    // to Dynamic Extensions
+    install
+}
+
+dependencies {
+    alfrescoProvided "org.alfresco.surf:spring-webscripts:6.20"
+    alfrescoProvided "org.alfresco:alfresco-data-model:8.25"
+    alfrescoProvided "org.alfresco:alfresco-repository:7.43"
+
+    // Note: SLF4J is already included in Alfresco, this is just included to demonstrate a custom installBundle task
+    install 'org.slf4j:slf4j-api:1.7.26'
+
+
+    /*
+    Following dependencies are not required to build the bundle or amp, but are used to setup an Alfresco for
+    demo purposes.
+    See https://github.com/xenit-eu/alfresco-docker-gradle-plugin for more information
+    */
+
+    baseAlfrescoWar "org.alfresco:content-services-community:${alfrescoVersion}@war"
+
+    // Include the Dynamic Extensions runtime AMP in Alfresco
+    alfrescoAmp "eu.xenit:alfresco-dynamic-extensions-repo-${dynamicExtensionsAlfrescoVersion}:${dynamicExtensionsVersion}@amp"
+
+    // Include our custom module in Alfresco
+    alfrescoDE this.project
```

Don't depend on the project itself, but depend on the `jar` task here.
```suggestion
    alfrescoDE jar
```

---

Apparently that doesn't seem to work...
```
FAILURE: Build failed with an exception.

* Where:
Build file '/home/daan/ws/xenit/de/example-dynamic-extension/gradle-with-plugin/build.gradle' line: 70

* What went wrong:
A problem occurred evaluating root project 'example-dynamic-extensions-module'.
> Cannot convert the provided notation to an object of type Dependency: task ':jar'.
  The following types/formats are supported:
    - Instances of Dependency.
    - String or CharSequence values, for example 'org.gradle:gradle-core:1.0'.
    - Maps, for example [group: 'org.gradle', name: 'gradle-core', version: '1.0'].
    - FileCollections, for example files('some.jar', 'someOther.jar').
    - Projects, for example project(':some:project:path').
    - ClassPathNotation, for example gradleApi().
  
  Comprehensive documentation on dependency notations is available in DSL reference for DependencyHandler type.

```

---

@vierbergenlars any other alternative we could try?

---

`jar.outputs` then (That should be a `FileCollection`)
Or `files(jar)` may also work

---

`files(jar)` :+1: